### PR TITLE
stanc3 2.29.2 (new formula)

### PR DESCRIPTION
**Pull request by ankane**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Stan-to-C++ compiler

### CODE REVIEW

File: Formula/stanc3.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      revision: "2c254b3b11a4dbb82395f016886b557201dad130"
+  license "BSD-3-Clause"
+
+  depends_on "ocaml" => :build
+  depends_on "opam" => :build
+
+  uses_from_macos "unzip" => :build
+
+  def install
+    Dir.mktmpdir("opamroot") do |opamroot|
```
Comment (SMillerDev) [inline Formula/stanc3.rb:16]: Since builds are already in a tmpdir, is this still needed?

Comment (ankane) [reply Formula/stanc3.rb]: Moved it to a directory in the buildpath (existing formulas with `opam` use either approach)

Comment (SMillerDev) [reply Formula/stanc3.rb]: Thanks, in that case this might also work:
```rb
mkdir "opamroot" do |opamroot|
```

Comment (ankane) [reply Formula/stanc3.rb]: Hmm, not sure why it fails with the buildpath approach (but don't know much about opam or dune), so maybe tmpdir is needed. Tried the `mkdir` approach as well but ran into the same issue.

**Review by SMillerDev — APPROVED** [review]
Thanks @ankane  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Comment by ankane** [comment]
Thanks @SMillerDev!

**Pull request closed without merging**